tool/tool.c

```diff
@@ -245,24 +299,28 @@ int main(int argc, char **argv) {
     dump_sector(&sector_buf.buffer);
   }
 
+  // Prep data for TSR/MDR overwrite.
+  prepare_hdd_data(&config, &sector_buf);
+
+  // The geometry can be different that what was reported above.
+  if (config.load_tsr || config.write) {
+    printf("Using Geometry: Cylinders=%i Heads=%i\n",
+      sector_buf.mbr.hdd_data.cylinders,
+      sector_buf.mbr.hdd_data.heads);
+  }
+
   // If we're loading the TSR, this will not return.
-  if (config.load_tsr) {
-    // Copy data from MBR
-    if (config.mbr_active) {
-      printf("Repatching with MBR data.\n");
+  if (config.load_tsr || config.write) {
```

Comment:
The more I'm thinking about this, the less I'm thinking it should update the TSR by default .. hmm. I can see this being an issue if the code is used on a different machine/dosbox to update an image when the TSR is not needed to actually access the image. So this likely needs a docs update rather than a default like this.